Hm 3

SunnatBakidjanov opened the pull request.

SunnatBakidjanov committed:
'Добавлены компоненты'

--Добавлен App
--Добавлен footer
--Добавлен header
--Добавлен layout
--Добавлен restaurants
--Контент из main.jsx перенесен в Restaurants.jsx

SunnatBakidjanov committed:
'Добавлены компоненты'

--Добавлен компонент Dish
--Добавлен компонент Menu
--Добавлен хук useCount

SunnatBakidjanov committed:
'Добавлены негативные условия в Dish'

SunnatBakidjanov committed:
'Поправлена логика в Dish'

SunnatBakidjanov committed:
'Добавлены новые компоненты'

--Добавлен компонент ReviewItem
--Добалвен компонент Reviews

SunnatBakidjanov committed:
'Добавлен новый компонент'

--Добавлен компонент RestaurantItem

SunnatBakidjanov committed:
'Добавлена логика табов + Валидация вынесена'

--Добален файл RestaurantTabs.jsx
--Добавлен файл TabsButtons.jsx
--Логика валидации вынесена в отдельные js файлы
--Добавлен хук useTabs.js
--Изменна логика в Restaurants.jsx

SunnatBakidjanov committed:
'Небольшие правки'

zubkov7 commented on src/components/restaurants/useTabs.js:
```diff
@@ -0,0 +1,16 @@
+import { useState } from "react";
+
+export const useTabs = restaurants => {
+    const [activeTab, setActiveTab] = useState(restaurants[0]?.id);
```
нужно отразить в названии переменной, что хранится именно id

zubkov7 commented on src/components/restaurants/Restaurants.jsx:
```diff
@@ -0,0 +1,39 @@
+export const Restaurants = () => {
+    const { activeTab, handleTabClick } = useTabs(restaurants);
+
+    const filteredRestaurants = restaurants.filter(restaurant => restaurant.id === activeTab);
```
для отрисовки 1 ресторана подойдет метод find. Сейчас вы пробегаетесь по массиву и потом опять отрисовываете массив

zubkov7 commented on src/components/restaurants/Restaurants.jsx:
```diff
@@ -0,0 +1,39 @@
+            <ul className='restaurants-list'>
+                {filteredRestaurants.length > 0 ? (
+                    filteredRestaurants.map(restaurant => {
+                        const validatedRestaurant = validateRestaurant(restaurant);
```
сейчас с валидацией получась связанная логика, лучше не переписывать массив на строку. Пусть сам компонент определяет, что отрисовать, если массив пустой

SunnatBakidjanov committed:
'Правки'

--Отразил названии переменной и функции что хранится id
--Для отрисовки ресторанов используется метод find заместо filter
--Изменил проверку на наличие ресторана

zubkov7 commented on src/components/restaurants/Restaurants.jsx:
```diff
@@ -5,33 +5,23 @@ import { RestaurantTabs } from "./restaurantTabs/RestaurantTabs";
-                    })
-                ) : (
-                    <li className='restaurants-item'>
-                        <p>Тут должен быть какой-то ресторан, но его нет. Попробуйте выбрать другой.</p>
+                {findRestaurant ? (
+                    <li key={findRestaurant.id}>
+                        <RestaurantItem {...validateRestaurant(findRestaurant)} />
                     </li>
+                ) : (
+                    <p>Нет подходящего ресторана для отображения</p>
                 )}
             </ul>
```
сейчас это уже не список, отображаем просто 1 ресторан

zubkov7 commented:
обычно данные не валидируют таким образом, компонент проверяет их и рисует соответствующий UI
